packages/relay-compiler/core/RelaySourceModuleParser.js

```diff
@@ -88,13 +88,15 @@ module.exports = (tagFinder: GraphQLTagFinder) => {
     });
   }
 
-  function getFileFilter(baseDir: string): FileFilter {
+  function getFileFilterDefault(baseDir: string): FileFilter {
     return (file: File) => {
       const text = fs.readFileSync(path.join(baseDir, file.relPath), 'utf8');
       return text.indexOf('graphql') >= 0;
     };
   }
 
+  const getFileFilter = fileFilter ? fileFilter : getFileFilterDefault;
```

Comment:
What do you think of changing this to always read the file and yield the contents to the plugin, rather than having the plugin read the file?

I also feel like not making this optional but having each plugin implement their version would make the API clearer, although it would technically be a breaking change. Not sure in how much the Relay team would care about that, especially as there’s really only the TS plugin that’s currently in existence (afaik). @jstejada what say you?